R-package/tests/testthat/test_weighted_loss.R

```diff
@@ -0,0 +1,66 @@
+context("Case weights are respected")
```

Comment:
This test looks good to me, thank you! This might be the first unit test we've added that covers weighted training at all in the R package, so really appreciate it.

It's ok with me for this to live in a new test file as you've set it up. `test_basic.R` (where most of the other `lgb.train()` tests live) has gotten kind of big, and there are already uses of `lgb.train()` outside of that file in their own files (such as https://github.com/microsoft/LightGBM/blob/c629cb0b6bd2830894b710cbd4d8241b82ac3105/R-package/tests/testthat/test_learning_to_rank.R or https://github.com/microsoft/LightGBM/blob/c629cb0b6bd2830894b710cbd4d8241b82ac3105/R-package/tests/testthat/test_custom_objective.R).